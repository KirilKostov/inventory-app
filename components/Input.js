import React from "react";
import { TextInput, StyleSheet } from "react-native";
import Colors from "../constants/colors";

export const Input = (props) => {
  return <TextInput {...props} style={{ ...styles.input, ...props.style }} />;
};

const styles = StyleSheet.create({
  input: {
    height: 40,
    borderWidth: 1,
    borderColor: Colors.gray,
    borderRadius: 4,
    marginVertical: 10,
    textAlign: "center",
  },
});
