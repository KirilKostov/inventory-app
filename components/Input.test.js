import React from "react";
import renderer from "react-test-renderer";

import { Input } from "./Input";

describe("<Input />", () => {
  it("renders correctly", () => {
    const testRenderer = renderer.create(<Input />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
  it("turns autoCorrect off", () => {
    const testRenderer = renderer.create(<Input autoCorrect={false} />);
    const testInstance = testRenderer.root;
    expect(testInstance.props.autoCorrect).toBe(false);
  });
  it("displays the correct placeholder", () => {
    const testRenderer = renderer.create(<Input placeholder="barcode" />);
    const testInstance = testRenderer.root;
    expect(testInstance.props.placeholder).toBe("barcode");
  });
});
