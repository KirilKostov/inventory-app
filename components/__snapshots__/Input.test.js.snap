// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  style={
    Object {
      "borderColor": "#78757c",
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 40,
      "marginVertical": 10,
      "textAlign": "center",
    }
  }
  underlineColorAndroid="transparent"
/>
`;
